src/main/scala/nl/knaw/dans/easy/export/JSON.scala

```diff
@@ -0,0 +1,93 @@
+/** *****************************************************************************
+  * Copyright 2015 DANS - Data Archiving and Networked Services
+  *
+  * Licensed under the Apache License, Version 2.0 (the "License");
+  * you may not use this file except in compliance with the License.
+  * You may obtain a copy of the License at
+  *
+  * http://www.apache.org/licenses/LICENSE-2.0
+  *
+  * Unless required by applicable law or agreed to in writing, software
+  * distributed under the License is distributed on an "AS IS" BASIS,
+  * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  * See the License for the specific language governing permissions and
+  * limitations under the License.
+  * *****************************************************************************/
+package nl.knaw.dans.easy.export
+
+import java.io.File
+
+import org.json4s.JsonAST
+import org.json4s.JsonAST.JObject
+import org.json4s.JsonDSL._
+import org.json4s.native.JsonMethods._
+
+import scala.util.{Failure, Try}
+import scala.xml.{Elem, Node}
+
+object JSON {
+
+  /** Creates a JSON structure while replacing some fedora ids with SDO directories on the flight.
+    * 'Some' being objects of an EASY-dataset.
+    *
+    * @param sdoDir a directory for the staged digital object
+    * @param datastreams subsection of foXML
+    * @param relsExt the content of the RESL-EXT datastream
+    * @param s configuration required to construct an sdoDir from a fedora-id
+    * @return a string representation of a json object
+    */
+  def apply(sdoDir: File,
+            datastreams: Seq[Node],
+            relsExt: Elem
+           )(implicit s: Settings): Try[String] = Try {
+
+    val descriptionNode = (relsExt \ "Description").head
+    val objectId = descriptionNode.attribute(rdf, "about").get.head.text.replaceAll("[^/]*/", "")
+    val relations = descriptionNode.descendant.filter(_.label != "#PCDATA")
```

Comment:
@DANS-KNAW/easy Special attention for XML handling for an educational purpose.

I got an headache from code I found elsewhere, even when inspecting the scala-doc of the used methods. An example of such code:

    case audience@Elem("dct", "audience", attribs, scope, _*) if audience.text.trim() == setting

So I experimented with a scala-worksheet and came up with the code applied in the three lines above, but also in the methods below and in the method `exportObject` of the main class `EasyExportDataset`. See also the tests in `JsonSpec` for examples of the parsed XML.

Because of namespaces I was not able to apply the neat "\@" above and had to use the verbose method `attribute`. Note how I used the scala methods filter and filterNot. It all gets pretty close to the power of real XPath. Note also that `.get`, `.head` and `.last` are dangerous and therefore are wrapped in a try.